pkg/systemconfig/syncer.go

```diff
@@ -76,17 +86,49 @@ func (s *SystemConfigSyncer) StoreImageRegistryConf(allowedRegistries []string,
 			signature.NewPRInsecureAcceptAnything(),
 		}
 		rc.Blocked = nil
+
+		for _, mirror := range rc.Mirrors {
+			if allowedRegistriesMap[mirror.Location] {
+				mirror.Insecure = &trueValue
+			}
+		}
+	}
+
+	// Create a map for quick lookup to reduce the time complexity
+	blockedRegistriesMap := make(map[string]bool)
+	for _, registry := range blockedRegistries {
+		blockedRegistriesMap[registry] = true
 	}
+
 	for _, registry := range blockedRegistries {
 		rc := s.registriesConfContent.getRegistryConfOrCreate(registry)
 		rc.Blocked = &trueValue
 		s.policyConfContent.Transports[dockerTransport][registry] = signature.PolicyRequirements{
 			signature.NewPRReject(),
 		}
+
+		for _, mirror := range rc.Mirrors {
+			if blockedRegistriesMap[mirror.Location] {
+				mirror.Insecure = &trueValue
+			}
+		}
+	}
+
+	// Create a map for quick lookup to reduce the time complexity
+	insecureRegistryMap := make(map[string]bool)
+	for _, registry := range insecureRegistries {
+		insecureRegistryMap[registry] = true
 	}
+
 	for _, registry := range insecureRegistries {
 		rc := s.registriesConfContent.getRegistryConfOrCreate(registry)
 		rc.Insecure = &trueValue
+
+		for _, mirror := range rc.Mirrors {
+			if insecureRegistryMap[mirror.Location] {
+				mirror.Insecure = &trueValue
+			}
+		}
 	}
 	s.registriesConfContent.cleanupAllRegistryConfIfEmpty()
```

Comment:
I see some value in having an increased in-memory complexity vs an increased processing complexity, but we are having this snippet of code copied over 4 times.

Also, we already have a lookup map to support this 'faster' lookup strategy.

Can we just do the following after `cleanupAllRegistryConfIfEmpty()` call, once the comment in https://github.com/openshift/multiarch-tuning-operator/pull/177/files#r1705717318 for mirrorsFor is addressed?:

```suggestion
	s.registriesConfContent.cleanupAllRegistryConfIfEmpty()
	for _, rc := range s.registriesConfContent.Registries {
		if len(rc.Mirrors) > 0 {
			rc.Mirrors = s.mirrorsFor(rc.mirrors, rc.mirrors[0].pullType)
		}
	}
```